### PR TITLE
DefaultEventHandler should not fire events if the handler is being closed

### LINKED ISSUE #578: DefaultEventHandler should not fire events if the handler is being closed

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #578

### CODE REVIEW

**Review by csviri — APPROVED** [review]

File: operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/DefaultEventHandlerTest.java [hunk 1]
```diff
@@ -224,6 +232,14 @@ public void scheduleTimedEventIfInstructedByPostExecutionControl() {
         .scheduleOnce(any(), eq(testDelay));
   }
 
+  @Test
+  public void doNotFireEventsIfClosing() {
+    defaultEventHandler.close();
+    defaultEventHandler.handleEvent(prepareCREvent());
+
+    verify(eventDispatcherMock, timeout(50).times(1)).handleExecution(any());
```
Comment (metacosm) [inline operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/DefaultEventHandlerTest.java:240]: Shouldn't we not dispatch at all if the handler is closed?

Comment (lburgazzoli) [reply operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/DefaultEventHandlerTest.java:240]: mh yes, don't know why I ended up with this test

Comment (lburgazzoli) [reply operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/DefaultEventHandlerTest.java:240]: will fix it later today

Comment (lburgazzoli) [reply operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/DefaultEventHandlerTest.java]: @metacosm fixed

**Review by metacosm — APPROVED** [review]
LGTM

**Pull request merged**